test/dsp.jl

```diff
@@ -31,6 +32,136 @@ import DSP: filt, filt!, deconv, conv, conv2, xcorr
     @test_throws ArgumentError filt!([1, 2], [1], [1], [1])
 end
 
+
+@testset "conv" begin
+        
+    @testset "conv-1D" begin
+        # Convolution
+        a = [1, 2, 1, 2]
+        b = [1, 2, 3]
+        expectation = [1, 4, 8, 10, 7, 6]
+        im_expectation = [1, 3, 6, 6, 5, 3]
+        @test conv(a, b) == expectation
+        @test conv(complex.(a, 1), complex.(b)) == complex.(expectation,
+                                                           im_expectation)
+        fa = convert(Array{Float64}, a)
+        fb = convert(Array{Float64}, b)
+        fexp = convert(Array{Float64}, expectation)
+        im_fexp = convert(Array{Float64}, im_expectation)
+        @test conv(fa, fb) ≈ fexp
+        @test conv(complex.(fa, 1.), complex.(fb)) ≈ complex.(fexp, im_fexp)
+        
+        @test conv(fa, b) ≈ fexp
+        @test conv(fb, a) ≈ fexp
+        @test_broken conv(complex.(fa, 1.), complex.(b)) ≈ complex.(fexp, im_fexp)
```

Comment:
Good catch on this, I had no idea that these sorts of method errors were lurking.